src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Solicitor.java

```diff
@@ -94,4 +104,52 @@ public boolean hasOrgId() {
         }
         return false;
     }
+
+    public String getAddress() {
+        if ((YesOrNo.NO == getAddressOverseas() || null == getAddressOverseas()) && (this.address != null && !this.address.equals(""))) {
```

Comment:
It would be nice if we could turn this into a guard clause to avoid wrapping the entire method in a block:
```suggestion
        if (getAddressOverseas() == YesOrNo.YES || this.address == null || this.address.equals("")) {
          return this.address;
        }
```
I think we would also need to check for the match against `ukPostcodeRegex` though

Reply:
Yeah we'd still have to run the matcher and default to returning the field as provided if there's no match.  I don't know if it would provide much benefit in this instance.